### PR TITLE
SUS-1312 (Chat) New user badges

### PR DESCRIPTION
**Pull request by gitkuba**
[SUS-1312](https://wikia-inc.atlassian.net/browse/SUS-1312)

Related change in Chat server: https://github.com/Wikia/chat/pull/39

### CODE REVIEW

File: extensions/wikia/Chat2/js/views/views.js [hunk 1]
```diff
@@ -211,8 +211,9 @@ var UserView = Backbone.View.extend({
 	render: function(){
 		//$().log("ABOUT TO RENDER THIS USER: " + JSON.stringify(this.model));
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/views/views.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: extensions/wikia/Chat2/js/views/views.js [hunk 2]
```diff
@@ -224,15 +225,16 @@ var UserView = Backbone.View.extend({
...
-		}
-
-		if(this.model.get('isStaff') === true){
-			$(this.el).addClass('staff');
-		}
+		// Check user groups so that kick-ban links don't show up, etc. & user gets a proper badge
+		$(this.el).addClass(groups.join(' '));
 
+		// if(this.model.get('isModerator') === true){
+		// 	$(this.el).addClass('chatmoderator');
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/views/views.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: extensions/wikia/Chat2/js/views/views.js [hunk 3]
```diff
@@ -580,7 +582,7 @@ var NodeChatUsers = Backbone.View.extend({
 		// admin actions
 		if (actions.admin && actions.admin.length) {
 			var adminActions = ul.clone().addClass('admin-actions');
-			for (var i in actions.admin) { 
+			for (var i in actions.admin) {
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/views/views.js:581]: MAJOR Rename "i" as this name is already used in declaration at line 551. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2814)
MAJOR Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)

File: extensions/wikia/Chat2/js/views/views.js [hunk 4]
```diff
@@ -580,7 +582,7 @@ var NodeChatUsers = Backbone.View.extend({
...
 				var action = actions.admin[i],
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/views/views.js:582]: MAJOR Rename "action" as this name is already used in declaration at line 552. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2814)

File: extensions/wikia/Chat2/js/views/views.js [hunk 5]
```diff
@@ -224,15 +225,16 @@ var UserView = Backbone.View.extend({
...
-			$(this.el).addClass('staff');
-		}
+		// Check user groups so that kick-ban links don't show up, etc. & user gets a proper badge
+		$(this.el).addClass(groups.join(' '));
 
+		// if(this.model.get('isModerator') === true){
+		// 	$(this.el).addClass('chatmoderator');
+		// }
+		//
+		// if(groups.indexOf("staff") != -1){
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/views/views.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: extensions/wikia/Chat2/js/views/views.js [hunk 6]
```diff
@@ -580,7 +582,7 @@ var NodeChatUsers = Backbone.View.extend({
 		// admin actions
 		if (actions.admin && actions.admin.length) {
 			var adminActions = ul.clone().addClass('admin-actions');
-			for (var i in actions.admin) { 
+			for (var i in actions.admin) {
 				var action = actions.admin[i],
 					template = _.template( $('#user-action-'+action+'-template').html() );
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/views/views.js:583]: MAJOR Rename "template" as this name is already used in declaration at line 553. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2814)

File: extensions/wikia/Chat2/js/views/views.js [hunk 7]
```diff
@@ -224,15 +225,16 @@ var UserView = Backbone.View.extend({
...
-		}
+		// Check user groups so that kick-ban links don't show up, etc. & user gets a proper badge
+		$(this.el).addClass(groups.join(' '));
 
+		// if(this.model.get('isModerator') === true){
+		// 	$(this.el).addClass('chatmoderator');
+		// }
+		//
+		// if(groups.indexOf("staff") != -1){
+		// 	$(this.el).addClass('staff');
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/views/views.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 157 issues
* CRITICAL 1 critical
* MAJOR 121 major
* MINOR 21 minor
* INFO 14 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [controllers.js#L101](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/js/controllers/controllers.js#L101): End this switch case with an unconditional break, continue, return or throw statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANonEmptyCaseWithoutBreak)
1. MAJOR [ChatAjax.class.php#L28](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/ChatAjax.class.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [controllers.js#L31](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)
1. MAJOR [controllers.js#L68](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/js/controllers/controllers.js#L68): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L141](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/js/controllers/controllers.js#L141): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L141](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/js/controllers/controllers.js#L141): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L145](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/js/controllers/controllers.js#L145): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L177](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/js/controllers/controllers.js#L177): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L217](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/js/controllers/controllers.js#L217): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L219](https://github.com/Wikia/app/blob/497ab07c772720ee86174784ea59b7de06b8ef5f/extensions/wikia/Chat2/js/controllers/controllers.js#L219): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

File: extensions/wikia/Chat2/js/views/views.js [hunk 8]
```diff
@@ -224,16 +223,15 @@ var UserView = Backbone.View.extend({
...
 
-		// If this is a chat moderator, add the chat-mod class so that kick-ban links don't show up, etc.
+		// Check user groups so that kick-ban links don't show up, etc. & user gets a proper badge
 		if(this.model.get('isModerator') === true){
-			$(this.el).addClass('chat-mod');
+			$(this.el).addClass('chatmoderator');
 		}
 
-		if(this.model.get('isStaff') === true){
+		if(groups.indexOf("staff") != -1){
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/views/views.js]: MAJOR Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

File: extensions/wikia/Chat2/js/views/views.js [hunk 9]
```diff
@@ -224,16 +223,15 @@ var UserView = Backbone.View.extend({
...
 			$(this.el).addClass('staff');
 		}
 
-
 		// If the user is away, add a certain class to them, if not, remove the away class.
 		if(this.model.get('statusState') == STATUS_STATE_AWAY){
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/views/views.js]: MAJOR Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

File: extensions/wikia/Chat2/js/controllers/controllers.js [hunk 10]
```diff
@@ -99,7 +99,8 @@ var NodeChatSocketWrapper = $.createClass(Observable, {
 				this.forceReconnect();
 				break;
 			case 'initial':
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/controllers/controllers.js]: CRITICAL End this switch case with an unconditional break, continue, return or throw statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANonEmptyCaseWithoutBreak)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 151 issues
* CRITICAL 1 critical
* MAJOR 115 major
* MINOR 21 minor
* INFO 14 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ChatAjax.class.php#L28](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/ChatAjax.class.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [controllers.js#L31](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)
1. MAJOR [controllers.js#L68](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/js/controllers/controllers.js#L68): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L142](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/js/controllers/controllers.js#L142): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L142](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/js/controllers/controllers.js#L142): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L146](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/js/controllers/controllers.js#L146): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L178](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/js/controllers/controllers.js#L178): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L218](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/js/controllers/controllers.js#L218): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L220](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/js/controllers/controllers.js#L220): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L233](https://github.com/Wikia/app/blob/caeb4b37a73843c03cafb1ae457cbfc084bbc29c/extensions/wikia/Chat2/js/controllers/controllers.js#L233): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 151 issues
* CRITICAL 1 critical
* MAJOR 115 major
* MINOR 21 minor
* INFO 14 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ChatAjax.class.php#L28](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/ChatAjax.class.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [controllers.js#L31](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)
1. MAJOR [controllers.js#L68](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/js/controllers/controllers.js#L68): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L142](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/js/controllers/controllers.js#L142): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L142](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/js/controllers/controllers.js#L142): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L146](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/js/controllers/controllers.js#L146): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L178](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/js/controllers/controllers.js#L178): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L218](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/js/controllers/controllers.js#L218): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L220](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/js/controllers/controllers.js#L220): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L233](https://github.com/Wikia/app/blob/8761000a1d066ba20efd64e56d185998c996326b/extensions/wikia/Chat2/js/controllers/controllers.js#L233): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 151 issues
* MAJOR 116 major
* MINOR 21 minor
* INFO 14 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ChatAjax.class.php#L28](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/ChatAjax.class.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [controllers.js#L31](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)
1. MAJOR [controllers.js#L68](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/js/controllers/controllers.js#L68): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L94](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/js/controllers/controllers.js#L94): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [controllers.js#L140](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L140](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L144](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/js/controllers/controllers.js#L144): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L176](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/js/controllers/controllers.js#L176): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L216](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/js/controllers/controllers.js#L216): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L218](https://github.com/Wikia/app/blob/cbd8f54a433b75310f09889ce763fd64a7943e7c/extensions/wikia/Chat2/js/controllers/controllers.js#L218): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

File: extensions/wikia/Chat2/ChatAjax.class.php [hunk 11]
```diff
@@ -66,13 +66,14 @@ static public function getUserInfo() {
...
 			'since' => '',
+			'groups' => $user->getGroups(),
 
 			// Extra wg variables that we need.
 			'wgCityId' => $wgCityId,
 			'wgServer' => $wgServer,
 			'wgArticlePath' => $wgArticlePath
 		];
-
+wfDebug('CHAT: ' . json_encode($res['groups']) . "\n");
```
Comment (macbre) [inline extensions/wikia/Chat2/ChatAjax.class.php]: Intending, please

**Review by macbre — CHANGES_REQUESTED** [review]
See above (including `!==` comments from Mr @WikiaTech)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 149 issues
* MAJOR 114 major
* MINOR 21 minor
* INFO 14 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ChatAjax.class.php#L28](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/ChatAjax.class.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [controllers.js#L31](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)
1. MAJOR [controllers.js#L68](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/js/controllers/controllers.js#L68): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L94](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/js/controllers/controllers.js#L94): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [controllers.js#L140](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L140](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L144](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/js/controllers/controllers.js#L144): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L176](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/js/controllers/controllers.js#L176): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L216](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/js/controllers/controllers.js#L216): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L218](https://github.com/Wikia/app/blob/4764d95db364c4cd16623625b99b168658e8991d/extensions/wikia/Chat2/js/controllers/controllers.js#L218): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

**Review by macbre — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 149 issues
* MAJOR 114 major
* MINOR 21 minor
* INFO 14 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ChatAjax.class.php#L28](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/ChatAjax.class.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [controllers.js#L31](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)
1. MAJOR [controllers.js#L68](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/js/controllers/controllers.js#L68): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L94](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/js/controllers/controllers.js#L94): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [controllers.js#L140](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L140](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L144](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/js/controllers/controllers.js#L144): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L176](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/js/controllers/controllers.js#L176): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [controllers.js#L216](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/js/controllers/controllers.js#L216): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [controllers.js#L218](https://github.com/Wikia/app/blob/78a69e1b0e92e7921e573030d9fbd538fc3bade8/extensions/wikia/Chat2/js/controllers/controllers.js#L218): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

**Review by macbre — APPROVED** [review]

**Pull request merged**